### PR TITLE
Support for positional arguments

### PR DESCRIPTION
**Pull request by gingerwizard**
Closes https://github.com/ClickHouse/clickhouse-go/issues/542

Benchmark suggests this is actually more performant than numeric.

### CODE REVIEW

File: bind.go [hunk 1]
```diff
@@ -35,32 +35,70 @@ func Named(name string, value interface{}) driver.NamedValue {
...
-var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+func bindPositional(tz *time.Location, query string, args ...interface{}) (_ string, err error) {
+	var (
+		unbind = make(map[int]struct{})
+		params = make([]string, len(args))
+	)
+	for i, v := range args {
+		if fn, ok := v.(std_driver.Valuer); ok {
+			if v, err = fn.Value(); err != nil {
+				return "", nil
```
Comment (genzgd) [inline bind.go:82]: I don't understand why we return an empty query string and no error here?  Won't that cause weirdness and confusion in the caller?

File: bind.go [hunk 2]
```diff
@@ -35,32 +35,70 @@ func Named(name string, value interface{}) driver.NamedValue {
...
-	return bindNumeric(tz, query, args...)
+	if haveNumeric {
+		return bindNumeric(tz, query, args...)
+	}
+	return bindPositional(tz, query, args...)
 }
 
-var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+func bindPositional(tz *time.Location, query string, args ...interface{}) (_ string, err error) {
+	var (
```
Comment (genzgd) [inline bind.go:75]: Can this be simplified just by using a param array?  Since we don't have string keys just indexes seem sufficient.

**Review by genzgd — APPROVED** [review]

File: bind.go [hunk 3]
```diff
@@ -35,32 +35,70 @@ func Named(name string, value interface{}) driver.NamedValue {
 	}
 }
 
+var bindNumericRe = regexp.MustCompile(`\$[0-9]+`)
+var bindPositionalRe = regexp.MustCompile(`[?]`)
```
[the remark this answers is not in the document]
Comment (gingerwizard) [reply bind.go:39]: +1

File: bind.go (hunk 1 above)
Comment (gingerwizard) [reply bind.go:82]: yes this was be consistent with the other methods..i'd need to dig into under what conditions this is used and why we handle it this way. I'll work on a PR to address all 3 methods.

**Pull request merged**